### ISSUE TITLE
Wrong type of file

### ISSUE THREAD

**Issue by coolswood** [billthecoder046/to_csv#14]
When I export file on android I get `.octet-stream` format, but expected `.csv`. That said, if you rename the file format afterwards, it opens and looks correct

**Comment by billthecoder046** [comment]
Hi coolswood. I had updated the package. Is it corrected now?

**Comment by coolswood** [comment]
@bilalsaeedjh Unfortunately the same. Here is my code

```
import 'package:to_csv/to_csv.dart' as exportCSV;

void exportSmerToCSV(BuildContext context, Box<SmerState> smerBox) {
  sendEvent('CSV_exported', {});

  List<List<String>> listOfLists = [];
  List<String> header = [];

  header.addAll([
    'Date',
    'diary.field.event'.tr(),
    'diary.field.thought'.tr(),
    'diary.field.body'.tr(),
    'diary.field.behavior'.tr(),
    'diary.field.conclusion'.tr(),
    'diary.field.response'.tr(),
    'believe.title'.tr(),
    'diary.field.distortions'.tr(),
    'diary.field.emotions'.tr(),
    'comparison.for'.tr(),
    'comparison.against'.tr(),
  ]);

  for (var smer in smerBox.values) {
    String locale = context.locale.toString();
    List<String> dataRow = [];

    dataRow.add(DateFormat('yMMMMd', locale).format(
        DateTime.fromMillisecondsSinceEpoch(
            smer.dt * 1000)));
    dataRow.add(smer.event ?? '');
    dataRow.add(smer.thoughts ?? '');
    dataRow.add(smer.feelInBody ?? '');
    dataRow.add(smer.behaviour ?? '');
    dataRow.add(smer.conclusion ?? '');
    dataRow.add(smer.solution ?? '');
    dataRow.add(smer.beliefLevel.toString());
    dataRow.add(smer.errorsList.map((e) => errorsListData[e]!.head.tr()).join(', '));
    dataRow.add(smer.emotionsList?.entries
        .map((e) => '${e.key.tr()}: ${e.value}')
        .join(', ') ??
        '');

    SmerComparison? comparison = smer.comparison;
    if (comparison != null) {
      String comparisonForItems = comparison.comparisonFor
          .map((item) => '${item.text}: ${item.score}')
          .join(', ');
      String comparisonAgainstItems = comparison.comparisonAgainst
          .map((item) => '${item.text}: ${item.score}')
          .join(', ');

      dataRow.add(comparisonForItems);
      dataRow.add(comparisonAgainstItems);
    } else {
      dataRow.addAll(['', '']);
    }

    listOfLists.add(dataRow);
  }

  exportCSV.myCSV(header, listOfLists, sharing: true);
}
```

**Comment by rahul69x** [comment]
I have same problem .octet-stream file extension saving. It was working before correctly. But after updating the plugin facing the issue.

**Comment by rahul69x** [comment]
```
library to_csv;

import 'dart:convert';

import 'package:file_saver/file_saver.dart';
import 'package:flutter/foundation.dart';
import 'package:csv/csv.dart';
import 'package:intl/intl.dart';
import 'package:share_plus/share_plus.dart';
import 'dart:io';
import 'package:universal_html/html.dart' as html;

 
Future myCSV(List<String> headerRow, List<List<String>> listOfListOfStrings,
    {bool sharing = false, String? fileName, String? fileTimeStamp}) async {
 
  print("***** Gonna Create cv");
  String givenFileName = "${fileName ?? 'item_export'}";

  DateTime now = DateTime.now();
 
  String formattedDate =
      fileTimeStamp ?? DateFormat('MM-dd-yyyy-HH-mm-ss').format(now);
 
  List<List<String>> headerAndDataList = [];
  headerAndDataList.add(headerRow);
  for (var dataRow in listOfListOfStrings) {
    headerAndDataList.add(dataRow);
  }

  String csvData = const ListToCsvConverter().convert(headerAndDataList);

 
  if (kIsWeb) {
    final bytes = utf8.encode(csvData);
    final blob = html.Blob([bytes]);
    final url = html.Url.createObjectUrlFromBlob(blob);
    final anchor = html.document.createElement('a') as html.AnchorElement
      ..href = url
 
      ..download = '$givenFileName$formattedDate.csv';
    html.document.body!.children.add(anchor);
    anchor.click();
    html.Url.revokeObjectUrl(url);
  } else if (Platform.isAndroid ||
      Platform.isIOS ||
      Platform.isWindows ||
      Platform.isMacOS) { 
    final bytes = utf8.encode(csvData);
    Uint8List bytes2 = Uint8List.fromList(bytes);
    MimeType type = MimeType.csv;
    if (sharing == true) {
      XFile xFile = XFile.fromData(bytes2, mimeType: "csv");
      await Share.shareXFiles([xFile], text: 'Csv File');
    } else {
      String formattedData = DateTime.now().millisecondsSinceEpoch.toString();
      String? unknownValue = await FileSaver.instance
          .saveAs(name: '$givenFileName$formattedDate.csv', bytes: bytes2, ext: 'csv', mimeType: type);
      print("Unknown value $unknownValue");
 
    }
  }
}

```

Change in the plugin code. I added the mime type as csv 

```
if (sharing == true) {
      XFile xFile = XFile.fromData(bytes2, mimeType: "csv");
      await Share.shareXFiles([xFile], text: 'Csv File');
    } 
```

**Comment by billthecoder046** [comment]
@rahul69x would appreciate if you request above changes in the recent updated version of to_csv 3.0.0.

**Comment by billthecoder046** [comment]
Resolved in to_csv: ^3.0.3
